added DAVE  - Design Analyse Visualize Export (or Difficult Analysis Very Easy)

RubendeBruin opened the pull request:
@conda-forge/help-python

This is the first contribution to conda-forge. Hope you will accept my contribution.

Tests:
The package is a book-keeping package to make the flow of information between various engineering packages and visualization packages easier for users. This means that there are many optional additional packages. I have tests for all of those locally but can not include them here (software like eg blender is needed to run them) so I have included only the minimal test.

Checklist:
- [X] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml" 
- [X] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [X] Source is from official source
- [X] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [X] If static libraries are linked in, the license of the static library is packaged.
- [X] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [X] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipes/DAVE```) and found some lint.

Here's what I've got...


For **recipes/DAVE**:

 * There are too few lines.  There should be one empty line at the end of the file.
 * about/license_family 'MPL' not allowed. Allowed families are AGPL,
APACHE, BSD, CC, GPL, GPL2, GPL3, LGPL, MIT, MOZILLA, NONE, OTHER,
PROPRIETARY, PSF, and PUBLIC-DOMAIN.
 * ``requirements: run: vtk >= 8.2.0`` should not contain a space between relational operator and the version, i.e. ``vtk >=8.2.0``

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipes/DAVE```) and found some lint.

Here's what I've got...


For **recipes/DAVE**:

 * There are too few lines.  There should be one empty line at the end of the file.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/DAVE```) and found it was in an excellent condition.

RubendeBruin commented:
@conda-forge/help-python
Linux and OSX builds work, but the win64 build seems to fail because libtiff 4.1.0 can not be obtained from https://conda.anaconda.org/conda-forge/win-64/libtiff-4.1.0-h21b02b4_4.tar.bz2

I do not need this dependancy to build or test, it comes with vtk. Is there a way to exclude this to temporary work around this issue?

![afbeelding](https://user-images.githubusercontent.com/34062862/71987293-38242a00-322e-11ea-81ef-c61be7c042c1.png)

RubendeBruin commented:
@conda-forge/help-python this is ready for review. For the time being I have removed VTK 8.2.0 from the requirements as work-around for the failing libtiff package.

RubendeBruin commented:
Hi @conda-forge/help-python , the package is now ready for review.

npavlovikj commented:
@RubendeBruin , when you create recipes for packages already available on "pypi", the preferred way to do that is with `conda-skeleton` (`conda-skeleton pypi usedave`), https://docs.conda.io/projects/conda-build/en/latest/user-guide/tutorials/build-pkgs-skeleton.html. This way, the skeleton of the recipe is there, but you will still need to do manual clean-up and edits. 
Can you please try that? I have few minor comments, that should be fixed when the skeleton is used.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipes/DAVE```) and found some lint.

Here's what I've got...


For **recipes/DAVE**:

 * Failed to even lint the recipe, probably because of a conda-smithy bug :cry:. This likely indicates a problem in your `meta.yaml`, though. To get a traceback to help figure out what's going on, install conda-smithy and run `conda smithy recipe-lint .` from the recipe directory.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipes/DAVE```) and found some lint.

Here's what I've got...


For **recipes/DAVE**:

 * Non noarch: python packages should have a python requirement without any version constraints.
 * Non noarch: python packages should have a python requirement without any version constraints.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/DAVE```) and found it was in an excellent condition.

RubendeBruin commented:
@npavlovikj , I used conda skeleton and made the following changes to the produces .yaml afterwards:

- changed package name to "DAVE" which is the name of the package. The name on pip was already taken.
- Changed license family from "OTHER" to "MOZILLA"
- Requirements, tests

However now the building fails on getting the .tar.gz files from pypi: 
https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=109879&view=logs&j=240f1fee-52bc-5498-a14a-8361bde76ba0&t=78a0099b-3f13-5ae9-c1f0-24f71d1eac20&l=96

what should I do?

xhochy commented on recipes/DAVE/meta.yaml:
```diff
@@ -0,0 +1,61 @@
+  script: "{{ PYTHON }} -m pip install . -vv"
+
+requirements:
+  build:
```
The whole build section can be omitted, this is only needed when you need compilers.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipes/DAVE```) and found some lint.

Here's what I've got...


For **recipes/DAVE**:

 * The recipe must have a `build/number` section.
 * Non noarch: python packages should have a python requirement without any version constraints.
 * Non noarch: python packages should have a python requirement without any version constraints.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/DAVE```) and found it was in an excellent condition.

scopatz commented on recipes/DAVE/meta.yaml:
```diff
@@ -0,0 +1,57 @@
+    - pyside2
+    - numpy
+    - scipy
+    - matplotlib
```
Can this be `matplotlib-base?`

RubendeBruin replied:
Checked and matplotlib-base works as well. Did not know it existed.... Thanks!

scopatz commented on recipes/DAVE/meta.yaml:
```diff
@@ -0,0 +1,57 @@
+    - pyscaffold
+  run:
+    - python >=3.5
+    - setuptools
```
Is setuptools really needed at runtime?

RubendeBruin replied:
Nope, you are right.  Removed it removed pyscaffold as well from run.

RubendeBruin commented:
@scopatz thanks for the comments. implemented, and ready to go

scopatz commented:
Thanks!
